backend/utils/checkExistentGap.js

```diff
@@ -0,0 +1,42 @@
+const getTimeInMinutes = require("./getTimeInMinutes");
+
+function checkExistentGap(appointmentList, doctorSchedule) {
+  //Now lets check if there is a gap of at least 30 minutes from the start of the doctors shift to the end
+  //Since the Appointments are ordered by the hour of the schedule this is more straight forward
+  for (let i = 0; i < appointmentList.length; i++) {
+    let appointmentDate = new Date(appointmentList[i].scheduleDate);
+
+    // Check the gap between the doctor's shift start and the first appointment of the day
+    if (
+      i === 0 &&
+      getTimeInMinutes(appointmentDate) -
+        getTimeInMinutes(new Date(doctorSchedule.startTime)) >
+        30
+    ) {
+      return true;
+    }
+    // Check the gap between the doctor's shift end and the last appointment of the day
+    if (
+      i === appointmentList.length - 1 &&
+      getTimeInMinutes(new Date(doctorSchedule.endTime)) -
+        getTimeInMinutes(appointmentDate) >
+        30
+    ) {
+      return true;
+    }
+
+    //Check the distance between the two consecutive appointments
+    if (
+      appointmentList[i + 1] &&
+      getTimeInMinutes(new Date(appointmentList[i + 1].scheduleDate)) -
+        getTimeInMinutes(appointmentDate) >
+        30
+    ) {
+      return true;
+    }
```

Comment:
Can we simplify the logic in this for loop? E.g. if we want to check the start / end time, we should put this logic outside the for loop because it doesn't have dependencies of the index `i`. This way, the code would be cleaner